Automattic/jetpack
# AI Extension: introduce Create with voice jetpack block

This PR creates the first implementation of the `Create with voice` block:

<img width="742" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/669a1d05-4125-4953-94ca-4f7f850d1a75">

<img width="692" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/1c69dcbd-5825-4132-b99d-cfa38ca48358">

<img width="360" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/93e730d7-f6c4-49e9-9655-f2f8ceba0bc2">

<img width="396" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/a4cd52be-fb28-436c-ae2f-87652cc0ca50">



### TS
Additionally, it moves TS files to the root of `extensions/` folder to increase its supports covering blocks (AI Assistant and this new one), plugins (AI Extension), etc...
Currently, we are duplicating files unnecessarily. 

Fixes #

## Proposed changes:

* AI Extension: add create-with-voice beta extension

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Create a Create with voice block instance
* Confirm the block is available only when `beta` extension is active.
* Take a look at the TS.  You can take a look how your IDE detect the globals, for instance:

<img width="797" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/af3dc847-37a1-4ac4-83c4-617f51da6215">
